### ISSUE TITLE
Add findings as issues and/or PR comments

### ISSUE THREAD

**Issue by flanakin** [microsoft/ApplicationInspector-Action#15]
I am curious about this, but I don't want to add the output to a file. I'd rather see the problems flagged on the PR they're addressing or filed as issues so they get fixed.

I would also like to see a full example of what the output looks like. Please add an example to the README.

**Comment by gfs** [comment]
Hi @Flanakin,

Application Inspector isn't designed to flag issues that need to be resolved. The intent is to be non-judgemental to audit what functionalities exist. Accordingly, flagging the locations where findings were detected in a PR is unlikely to be very actionable.  

For a similar workflow action that is intended to flag issue for resolution I can suggest the DevSkim-Action instead. DevSkim uses essentially the same rules engine as Application Inspector but is geared towards detecting common security issues and coding flaws. https://github.com/microsoft/DevSkim-Action

Ultimately, its up to you, if you do wish to have ApplicationInspector findings flagged in the PR, you can try adding the upload sarif command to your action workflow, and provide the path to the sarif the ApplicationInspector action generates.  If you have issues with that you can reach back out. You can add the below but replace results.sarif with the name of the sarif file you're generating.

```yml
   - uses: github/codeql-action/upload-sarif@v2
      with:
        sarif_file: results.sarif
```

For the second request, Sarif is a JSON based format, you may find it easiest to use a sarif viewer like https://github.com/microsoft/sarif-web-component to view the output sarif file, but the most interesting data in the output for you will likely be in `runs[0].results` of the output Sarif. Here's a sample of a single result:

```json
{
    "ruleId": "AI012800",
    "level": "note",
    "message": {
        "text": "Data: Serialization"
    },
    "locations": [
        {
            "physicalLocation": {
                "artifactLocation": {
                    "uri": ".\\AttackSurfaceAnalyzer\\Cli\\AttackSurfaceAnalyzerClient.cs",
                    "index": 0
                },
                "region": {
                    "startLine": 1295,
                    "startColumn": 77,
                    "endLine": 1295,
                    "endColumn": 92,
                    "snippet": {
                        "text": "SerializeObject"
                    }
                }
            }
        }
    ],
    "properties": {
        "tags": [
            "Data.Serialization"
        ],
        "AppInspector:Severity": "Critical"
    }
}
```
